### PR TITLE
DriveController set synchronous mode on each movement

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes

The DriveController enables synchronous mode in its constructor, which causes issues when trying to use an EncoderMotor individually; this PR will enable synchronous mode for each movement of the DriveController and then disable it, allowing to use an EncoderMotor alongside the DriveController.

Sample snippet:

```
from pitop import DriveController

d = DriveController()
d.forward(0.5)
d.stop()

left_motor = d.left_motor

# this doesn't work right now
left_motor.set_target_rpm(144)
```

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: packages/robotics/pitop/robotics/drive_controller.py [hunk 1]
```diff
@@ -79,8 +78,13 @@ def _set_synchronous_motor_movement_mode(self) -> None:
         )
         self.__mcu_device.write_byte(MotorSyncRegisters.CONFIG.value, sync_config)
 
+    def _unset_synchronous_motor_movement_mode(self) -> None:
+        self.__mcu_device.write_byte(MotorSyncRegisters.CONFIG.value, 0b0000000)
+
     def _start_synchronous_motor_movement(self) -> None:
+        self._set_synchronous_motor_movement_mode()
```
Comment (angusjfw) [inline packages/robotics/pitop/robotics/drive_controller.py:85]: instead of doing this sync mode either side of the 'start' we need to do it either side of setting the target rpms and start, eg doing it and the start and end of  _set_motor_speeds(

**Review by angusjfw — APPROVED** [review]
